File: project/frontend-services/libs/device-management/model/src/lib/model/Device.ts
export class Device {
  constructor(public id: string, public name: string, public downlink: string) {
  }

  static empty() {
    return new Device('', '', '');
  }

  clone() {
    return new Device(this.id, this.name, this.downlink);
  }

  same(device: Device) {
    return this.name == device.name && this.id == device.id;
  }
}

File: project/backend-services/device-commander/src/main/java/pt/sensae/services/device/commander/domain/device/Device.java
package pt.sensae.services.device.commander.domain.device;

public record Device(DeviceId id, DeviceName name, DeviceDownlink downlink) {

    public static Device empty(DeviceId id) {
        return new Device(id, new DeviceName(""), new DeviceDownlink(""));
    }
}
